tools/bazel

```diff
@@ -42,7 +42,7 @@ fi
 
 # IMPORTANT: if you update the version here, other parts of infrastructure might needs updating as well
 # (e.g. sanity checks, bazel toolchains etc.)
-VERSION=${OVERRIDE_BAZEL_VERSION:-4.2.1}
+VERSION=${OVERRIDE_BAZEL_VERSION:-5.0.0}
```

Comment:
note that the protoc and grpc_charp_plugin binaries aren't being built by bazel. See https://github.com/grpc/grpc/blob/3c558dd7962f2f2aea6b6a5759f65fa6f11db847/tools/internal_ci/macos/grpc_distribtests_csharp.sh for more info.